crates/proof-of-sql/src/proof_primitive/dory/public_parameters.rs

```diff
@@ -170,7 +170,7 @@ impl CanonicalDeserialize for PublicParameters {
     ) -> Result<Self, SerializationError> {
         // Deserialize max_nu (u64)
         let max_nu_u64 = u64::deserialize_with_mode(&mut reader, compress, validate)?;
-        let max_nu = max_nu_u64 as usize;
+        let max_nu: usize = max_nu_u64.try_into().unwrap_or(usize::MAX);
```

Comment:
Let's use `.map_err(|_| SerializationError::InvalidData)` here instead of `.unwrap_or`.